libsolidity/ast/Types.cpp

```diff
@@ -1864,6 +1873,45 @@ std::unique_ptr<ReferenceType> ArrayType::copyForLocation(DataLocation _location
 	return copy;
 }
 
+BoolResult ArraySliceType::isImplicitlyConvertibleTo(Type const& _other) const
+{
+	if (m_arrayType.location() == DataLocation::CallData && m_arrayType.isDynamicallySized() && m_arrayType == _other)
+		return true;
+	return (*this) == _other;
+}
+
+string ArraySliceType::richIdentifier() const
+{
+	return m_arrayType.richIdentifier() + "slice";
+}
+
+bool ArraySliceType::operator==(Type const& _other) const
+{
+	if (auto const* other = dynamic_cast<ArraySliceType const*>(&_other))
+		return m_arrayType == other->m_arrayType;
+	return false;
+}
+
+string ArraySliceType::toString(bool _short) const
+{
+	return m_arrayType.toString(_short) + " slice";
+}
+
+TypePointer ArraySliceType::encodingType() const
```

Comment:
Does that work?

Reply:
Well, since the array slice type can't be named at the moment, this should never actually be used... but if one ever was to encode or decode them, then at least dynamic calldata array slices can be encoded just like full dynamic calldata arrays at the moment... But maybe better to leave it at the default implementation for now...

Reply:
For `abi.encode`, the type does not have to be named.

Reply:
Yeah, true... that should actually just work like this then - I'll add tests for it.

Reply:
Hm... it seems like there is more to it - it still won't pass the type checker like this and codegen also has various assertions - I'll remove it and we can implement all this later.